swatch-metrics-hbi/src/test/java/com/redhat/swatch/hbi/events/processing/handlers/CreateUpdateHostHandlerTest.java

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.hbi.events.processing.handlers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.redhat.swatch.hbi.events.dtos.hbi.HbiHostCreateUpdateEvent;
+import com.redhat.swatch.hbi.events.normalization.FactNormalizer;
+import com.redhat.swatch.hbi.events.normalization.MeasurementNormalizer;
+import com.redhat.swatch.hbi.events.normalization.NormalizedEventType;
+import com.redhat.swatch.hbi.events.normalization.facts.RhsmFacts;
+import com.redhat.swatch.hbi.events.normalization.facts.SystemProfileFacts;
+import com.redhat.swatch.hbi.events.repository.HbiHostRelationship;
+import com.redhat.swatch.hbi.events.repository.HbiHostRelationshipRepository;
+import com.redhat.swatch.hbi.events.services.HbiHostRelationshipService;
+import com.redhat.swatch.hbi.events.test.helpers.HbiEventTestData;
+import com.redhat.swatch.hbi.events.test.helpers.HbiEventTestHelper;
+import com.redhat.swatch.hbi.events.test.helpers.SwatchEventTestHelper;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.candlepin.clock.ApplicationClock;
+import org.candlepin.subscriptions.json.Event;
+import org.candlepin.subscriptions.json.Event.HardwareType;
+import org.candlepin.subscriptions.json.Event.Sla;
+import org.candlepin.subscriptions.json.Event.Usage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
```

Comment:
It's much simpler to use `@QuarkusTest` instead of calling the constructor with all the arguments.